src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-overview-dashboard/rgw-overview-dashboard.component.ts

```diff
@@ -115,4 +135,132 @@ export class RgwOverviewDashboardComponent implements OnInit, OnDestroy {
       true
     );
   }
+
+  getSyncStatus() {
+    this.subscription = this.rgwMultisiteService.getSyncStatus().subscribe((data: any) => {
+      const result = data;
+      const [primaryZoneInfo, metadataSyncInfo] = result.split('metadata sync');
+      const [realmInfo, zgInfo, zoneInfo] = primaryZoneInfo.split('\n');
+      this.realm = this.getPrimaryZonedata(realmInfo);
+      this.zonegroup = this.getPrimaryZonedata(zgInfo);
+      this.zone = this.getPrimaryZonedata(zoneInfo);
+      this.metadataSyncInfo = metadataSyncInfo.split('data sync source')[0].trim();
+      if (this.metadataSyncInfo !== 'no sync (zone is master)') {
+        this.metadataSyncData = {};
+        const metadataSyncInfoArray = this.metadataSyncInfo.split('\n');
+        if (metadataSyncInfoArray.length > 0) {
+          const metadataSyncStatus = metadataSyncInfoArray[1];
+          let metadataFullSync = '';
+          let metadataIncrementalSync = '';
+          let metadataDataSyncStatus = '';
+          metadataSyncInfoArray.forEach((item: string) => {
+            if (item.includes('full sync') && item.endsWith('shards')) {
+              metadataFullSync = item;
+            }
+            if (item.includes('incremental sync')) {
+              metadataIncrementalSync = item;
+            }
+            if (item.includes('data is behind') || item.includes('data is caught up')) {
+              metadataDataSyncStatus = item;
+            }
+          });
+          this.metadataSyncData['syncstatus'] = metadataSyncStatus.trim();
+          this.metadataSyncData['fullSyncStatus'] = metadataSyncInfoArray;
+          this.metadataSyncData['fullSync'] = this.getShardsInfo(metadataFullSync.trim()).split(
+            '/'
+          );
```

Comment:
2 things that I never tire of repeating:
* Let's not put processing in the `subscribe()` section. That's blocking/synchronous and will impact performance/web responsivity. `subscribe()` is for assigning the value of the observable to a specific variable/member. Instead let's use `pipe(<complex_processing_goes_here>)`, which is asynchronous, non-blocking.
* Let's not do payload/data processing in the front-end, but in the back-end. I'll just point you to what I recently told @nizamial09: https://github.com/ceph/ceph/pull/52395#discussion_r1281601092. Additionally, if we do this, the above concern (complex logic) is no longer such, since the front-end doesn't have to deal with complex parsing/processing, just dumping/beautifying whatever comes from the back-end.